gtk/TorrentCellRenderer.cc

```diff
@@ -170,151 +143,161 @@ Glib::ustring getShortTransferString(
     return {};
 }
 
-Glib::ustring getShortStatusString(tr_torrent const* tor, tr_stat const* st, double uploadSpeed_KBps, double downloadSpeed_KBps)
+std::string getShortStatusString(
+    tr_torrent const* const tor,
+    tr_stat const* const st,
+    double uploadSpeed_KBps,
+    double downloadSpeed_KBps)
 {
-    Glib::ustring gstr;
-
     switch (st->activity)
     {
     case TR_STATUS_STOPPED:
-        gstr += st->finished ? _("Finished") : _("Paused");
-        break;
+        return st->finished ? _("Finished") : _("Paused");
 
     case TR_STATUS_CHECK_WAIT:
-        gstr += _("Queued for verification");
-        break;
+        return _("Queued for verification");
 
     case TR_STATUS_DOWNLOAD_WAIT:
-        gstr += _("Queued for download");
-        break;
+        return _("Queued for download");
 
     case TR_STATUS_SEED_WAIT:
-        gstr += _("Queued for seeding");
-        break;
+        return _("Queued for seeding");
 
     case TR_STATUS_CHECK:
-        gstr += gtr_sprintf(_("Verifying local data (%.1f%% tested)"), tr_truncd(st->recheckProgress * 100.0, 1));
-        break;
+        return fmt::format(
+            _("Verifying local data ({percent_done:.1}% tested)"),
+            fmt::arg("percent_done", tr_truncd(st->recheckProgress * 100.0, 1)));
 
     case TR_STATUS_DOWNLOAD:
     case TR_STATUS_SEED:
-        {
-            /* download/upload speed, ratio */
-            gstr += gtr_sprintf("%s  ", getShortTransferString(tor, st, uploadSpeed_KBps, downloadSpeed_KBps));
-            gstr += gtr_sprintf(_("Ratio: %s"), tr_strlratio(st->ratio));
-            break;
-        }
+        return fmt::format(
+            "{} {}",
+            getShortTransferString(tor, st, uploadSpeed_KBps, downloadSpeed_KBps),
+            fmt::format(_("Ratio: {percent}"), fmt::arg("percent", tr_strlratio(st->ratio))));
 
     default:
+        return {};
+    }
+}
+
+static std::optional<std::string> getErrorString(tr_stat const* st)
+{
+    switch (st->error)
+    {
+    case TR_STAT_TRACKER_WARNING:
+        return fmt::format(_("Tracker warning: '{warning}'"), fmt::arg("warning", st->errorString));
         break;
+
+    case TR_STAT_TRACKER_ERROR:
+        return fmt::format(_("Error: '{error}'"), fmt::arg("error", st->errorString));
```

Comment:
```suggestion
        return fmt::format(_("Tracker error: '{error}'"), fmt::arg("error", st->errorString));
```
Or maybe remove "Tracker " from "Tracker warning" above?

Reply:
f8f1b2b2c45456b2e028b4fa2d570e3b83931b66